### PR TITLE
chore: Make PropertyType.expected_type auto gen for dict and str

### PR DESCRIPTION
**Pull request by aahung**
### Checklist

- [ ] Adheres to the [development guidelines](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#development-guidelines)
- [ ] Add/update [transform tests](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#unit-testing-with-multiple-python-versions)
    - [ ] Using correct values
    - [ ] Using wrong values
- [ ] Add/update [integration tests](https://github.com/aws/serverless-application-model/blob/develop/INTEGRATION_TESTS.md)

### Examples?

Please reach out in the comments if you want to add an example. Examples will be 
added to `sam init` through [aws/aws-sam-cli-app-templates](https://github.com/aws/aws-sam-cli-app-templates).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by hoffa — APPROVED** [review]

File: samtranslator/model/__init__.py [hunk 1]
```diff
@@ -5,16 +5,16 @@
...
 from samtranslator.plugins import LifeCycleEvents
 from samtranslator.model.tags.resource_tagging import get_tag_list
 
 
 class PropertyType(object):
     """Stores validation information for a CloudFormation resource property.
 
-    The argument "expected_type" is only used by InvalidResourcePropertyTypeException
-    to generate an error message. When it is not provided,
+    The arribute "expected_type" is only used by InvalidResourcePropertyTypeException
```
Comment (xazhao) [inline samtranslator/model/__init__.py:16]: `attribute`

**Review by GavinZZ — COMMENTED** [review]
What's the benefit of this?

**Comment by aahung** [comment]
> What's the benefit of this?

so we don't have to pass `expected_type = xxx` maually

File: samtranslator/model/__init__.py (hunk 1 above)
Comment (aahung) [reply samtranslator/model/__init__.py:16]: lol

Comment (aahung) [reply samtranslator/model/__init__.py]: fixed

**Review by GavinZZ — APPROVED** [review]
Nice

**Comment by hoffa** [comment]
> What's the benefit of this?

Related to previous PR that added `PropertyType.optional_dict()`. Which is confusing since technically a property could be both required and `Optional`; `optional_dict` isn't `Optional[dict]`.

File: samtranslator/model/__init__.py [hunk 2]
```diff
@@ -5,16 +5,16 @@
 
 from samtranslator.intrinsics.resolver import IntrinsicsResolver
 from samtranslator.model.exceptions import ExpectedType, InvalidResourceException, InvalidResourcePropertyTypeException
```
Comment (xazhao) [inline samtranslator/model/__init__.py:7]: `ExpectedType` can be removed from import?

Comment (hoffa) [reply samtranslator/model/__init__.py:7]: If this is a valid comment, we should instead fix `pylint` so that it complains.

Comment (aahung) [reply samtranslator/model/__init__.py:7]: it is used at line 30

**Review by xazhao — APPROVED** [review]

**Pull request merged**